docs/OnClusterBuildInstructions.md

```diff
@@ -0,0 +1,345 @@
+    can verify this by running the following:
+
+    ```console
+    $ oc debug node/node-name
+    Starting pod/node-name-debug ...
+    To use host binaries, run `chroot /host`
+    Pod IP: 10.0.29.42
+    If you don't see a command prompt, try pressing enter.
+    sh-4.4# chroot /host
+    sh-5.1# rpm-ostree status
+    State: idle
+    Deployments:
+    * ostree-unverified-registry:quay.io/myorg/myrepo@sha256:324f17d18b197a951e7e11ea0b101836312191f92aefa0fc2ee240354cfbf7
+                    Digest: sha256:324f17d18b197a951e7e11ea0b101836312191f92aefa0fc2ee240354cfbf7
+                    Version: 414.92.202308210204-0 (2023-08-23T12:21:02Z)
+    ```
+
+## Applying new configurations
+
+1.  Adding a new MachineConfig to a layered MachineConfigPool will cause
+    a new image build to be performed.
+
+2.  After the new image is built, rolling out the new image will follow
+    the same process as rolling out new MachineConfigs does. The
+    difference is that the node will be rebased onto the new OS image
+    which contains the new MachineConfigs instead of having the
+    configs written directly to disk.
+
+# Interacting with the MOB/on-cluster builds
+
+## Viewing the Machine OS Builder logs
+
+One can run `$ oc logs -f "$(oc get pods -o name -l
+'k8s-app=machine-os-builder' -n openshift-machine-config-operator)" -n
+openshift-machine-config-operator` to view the Machine OS Builder pod
+logs, if the MOB is currently running.
+
+## Viewing the OS image build logs
+
+Depending on whether one is using the ImageBuildController or the
+PodBuildController, one may view the build logs in one of two ways:
+
+-   PodBuildController - `$ oc logs -f -n
+    openshift-machine-config-operator pod/build-<rendered
+    MachineConfig name>`. One may have to specify which containers’
+    logs to view. For example, one may need to add `-c image-build` to
+    view the image build process or `-c wait-for-done` to view the
+    digestfile ConfigMap creation process.
+
+-   ImageBuildController - `$ oc logs -f -n
+    openshift-machine-config-operator build/build-<rendered
+    MachineConfig name>`.
+
+## Viewing the rendered Dockerfile
+
+Whenever a build is currently in progress, the rendered Dockerfile may
+be viewed by running `$ oc get configmap/dockerfile-<rendered
+MachineConfig name> -n openshift-machine-config-operator`. However, if
+the build is successful, this ConfigMap will be deleted upon completion
+of the build process since it is no longer needed.
+
+## Viewing the rendered MachineConfig built into the final OS image
+
+Whenever a build is currently in progress, the rendered MachineConfig
+that is stored within the ConfigMap may be viewed by running `$ oc get
+configmap/mc-<rendered MachineConfig name> -n
+openshift-machine-config-operator -o json | jq -r
+'.data\["machineconfig.json.gz"\] | @base64d' | gunzip -`. However, if
+the build is successful, this ConfigMap will be deleted upon completion
```

Comment:
Another version that worked for me

```
$ oc get configmap/mc-rendered-layering-a5dfd76db247a1787fecb955e2280c8a -n openshift-machine-config-operator --template='{{index .data "machineconfig.json.gz" | base64decode}}'  | gunzip | jq |head
{
  "metadata": {
    "name": "rendered-layering-a5dfd76db247a1787fecb955e2280c8a",

```

Reply:
Yeah, this was another example of Pandoc escaping things that shouldn't have been escaped. I think another issue may be that I've got a Mac which has different versions of common userland utilities such as `gunzip`. I'll use your second option since it's the closest to the version that I have.